OpenRA.Game/Map/MapCache.cs

```diff
@@ -54,12 +54,20 @@ public MapCache(ModData modData)
 					? MapClassification.Unknown : Enum<MapClassification>.Parse(kv.Value);
 
 				IReadOnlyPackage package;
-				var optional = name.StartsWith("~");
+				var optional = name.StartsWith("~", StringComparison.Ordinal);
 				if (optional)
 					name = name.Substring(1);
 
 				try
 				{
+					// HACK: If this a reference to the support directory then we may need to create it
+					if (name.StartsWith("^", StringComparison.Ordinal))
+					{
+						var resolved = Platform.ResolvePath(name);
+						if (!File.Exists(resolved))
+							Directory.CreateDirectory(resolved);
```

Comment:
So why do a check at all? Do we not know if this should be a file or a directory?